### PR TITLE
GSYE-154: Add 'traded_energy' member to the 'Trade' dataclass

### PR DESCRIPTION
**Pull request by Hassan754**
In order to know the traded energy, we used to get it from `trade.offer_bid.energy`.
However,`energy` and `price` members of the bid and offer instances aren't the absolute source of truth now because if they have requirements they will be prioritized.
This PR adds `traded_energy` and `trade_price` members to the Trade object, so from now on we should read these values.

### CODE REVIEW

**Review by spyrostz — APPROVED** [review]
LGTM, but I assume there will be a followup PR about the gsy-e changes, right ?

**Comment by Hassan754** [comment]
> LGTM, but I assume there will be a followup PR about the gsy-e changes, right ?

@spyrostz right, I'm currently working on the gsye side

File: gsy_framework/data_classes.py [hunk 1]
```diff
@@ -388,6 +392,11 @@ def is_offer_trade(self) -> bool:
         """Check if the instance is an offer trade."""
         return isinstance(self.offer_bid, Offer)
 
+    @property
+    def trade_rate(self):
+        """Return the energy rate of the trade."""
+        return round(self.trade_price / self.traded_energy, 8)
```
Comment (hannesdiedrich) [inline gsy_framework/data_classes.py:398]: Please reuse the `DEFAULT_PRECISION` constant. Thanks!
```suggestion
        return round(self.trade_price / self.traded_energy, DEFAULT_PRECISION)
```

**Review by hannesdiedrich — DISMISSED** [review]
One comment, once adapted, LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**